iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -405,45 +380,63 @@ private Set<FileStatus> listForCommits(JobConf jobConf, String jobLocation) thro
    * @param executor The executor used to read the forCommit files
    * @param outputTable The table used for loading from the catalog
    */
-  private void commitTable(FileIO io, ExecutorService executor, OutputTable outputTable) {
+  private void commitTable(FileIO io, ExecutorService executor, OutputTable outputTable,
+                           List<JobContext> jobContextList) {
     String name = outputTable.tableName;
-    JobContext jobContext = outputTable.jobContext;
-    JobConf conf = jobContext.getJobConf();
     Properties catalogProperties = new Properties();
     catalogProperties.put(Catalogs.NAME, name);
     catalogProperties.put(Catalogs.LOCATION, outputTable.table.location());
     if (outputTable.catalogName != null) {
       catalogProperties.put(InputFormatConfig.CATALOG_NAME, outputTable.catalogName);
     }
-    Table table = Catalogs.loadTable(conf, catalogProperties);
 
+    Collection<DataFile> dataFiles = Lists.newArrayList();
+    Collection<DeleteFile> deleteFiles = Lists.newArrayList();
+
+    Table table = null;
+    String branchName = null;
+    boolean isOverWrite = false;
+
+    for (JobContext jobContext : jobContextList) {
+      JobConf conf = jobContext.getJobConf();
+      table = Optional.ofNullable(table).orElse(Catalogs.loadTable(conf, catalogProperties));
+
+      branchName = conf.get(InputFormatConfig.OUTPUT_TABLE_SNAPSHOT_REF);
+      isOverWrite = conf.getBoolean(InputFormatConfig.IS_OVERWRITE, false);
```

Comment:
Is it possible that some `jobContext` has `isOverwrite` `true` and some of them is `false`? If the answer is yes then this local variable has the last `jobContext`'s value. Is this expected?

Reply:
list of job context is grouped by table name to handle multi-insert statements after the split update/merge. so it should be the same